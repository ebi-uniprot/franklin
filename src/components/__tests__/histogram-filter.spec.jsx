import { render, fireEvent } from '@testing-library/react';
import { HistogramFilter } from '..';

describe('HistogramFilter component', () => {
  const onChange = jest.fn();

  let rendered;
  beforeEach(async () => {
    rendered = render(
      <HistogramFilter
        onChange={onChange}
        values={[...Array(10).keys()]}
        nBins={5}
        selectedRange={[0, 9]}
      />
    );
    const { findByTestId } = rendered;
    await findByTestId('histogram');
  });

  it('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onChange when text input is provided', () => {
    const { getByDisplayValue } = rendered;
    const startInput = getByDisplayValue('0');
    fireEvent.change(startInput, { target: { value: '1' } });
    expect(onChange).toHaveBeenCalledWith([1, 9]);
  });
});
