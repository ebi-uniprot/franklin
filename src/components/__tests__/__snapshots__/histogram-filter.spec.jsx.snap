// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistogramFilter component should render 1`] = `
<DocumentFragment>
  <div
    class="histogram-filter"
  >
    <div
      class="histogram-filter__histogram-rheostat-container"
    >
      <div
        class="histogram"
        data-testid="histogram"
      >
        <div
          class="histogram__bar-container"
          style="height: 50px;"
        >
          <div
            class="histogram histogram__bar histogram__bar--within-range"
            data-testid="histogram-bar"
            style="transform: scaleY(0.4); transition-delay: 0ms;"
          />
          <div
            class="histogram histogram__bar histogram__bar--within-range"
            data-testid="histogram-bar"
            style="transform: scaleY(0.4); transition-delay: 100ms;"
          />
          <div
            class="histogram histogram__bar histogram__bar--within-range"
            data-testid="histogram-bar"
            style="transform: scaleY(0.4); transition-delay: 200ms;"
          />
          <div
            class="histogram histogram__bar histogram__bar--within-range"
            data-testid="histogram-bar"
            style="transform: scaleY(0.4); transition-delay: 300ms;"
          />
          <div
            class="histogram histogram__bar histogram__bar--within-range"
            data-testid="histogram-bar"
            style="transform: scaleY(0.4); transition-delay: 400ms;"
          />
        </div>
      </div>
      <div
        class="rheostat rheostat_1"
      >
        <div
          class="DefaultBackground DefaultBackground_1 DefaultBackground_background__horizontal DefaultBackground_background__horizontal_2"
        />
        <div
          class="handleContainer handleContainer_1"
        >
          <button
            aria-disabled="false"
            aria-valuemax="9"
            aria-valuemin="0"
            aria-valuenow="0"
            class="DefaultHandle_handle DefaultHandle_handle_1 DefaultHandle_handle__horizontal DefaultHandle_handle__horizontal_2"
            data-handle-key="0"
            role="slider"
            style="left: 0%; position: absolute;"
            tabindex="0"
            type="button"
          />
          <button
            aria-disabled="false"
            aria-valuemax="9"
            aria-valuemin="0"
            aria-valuenow="9"
            class="DefaultHandle_handle DefaultHandle_handle_1 DefaultHandle_handle__horizontal DefaultHandle_handle__horizontal_2"
            data-handle-key="1"
            role="slider"
            style="left: 100%; position: absolute;"
            tabindex="0"
            type="button"
          />
        </div>
        <div
          class="DefaultProgressBar_progressBar DefaultProgressBar_progressBar_1 DefaultProgressBar_background__horizontal DefaultProgressBar_background__horizontal_2"
          style="left: 0%; width: 100%;"
        />
      </div>
    </div>
    <div
      class="histogram-filter__text-input-container"
    >
      <input
        max="9"
        min="0"
        style="width: 3ch;"
        type="number"
        value="0"
      />
      <input
        max="9"
        min="0"
        style="width: 3ch;"
        type="number"
        value="9"
      />
    </div>
  </div>
</DocumentFragment>
`;
